pkg/tsdb/prometheus/querydata/response.go

```diff
@@ -41,7 +38,7 @@ func (s *QueryData) parseResponse(ctx context.Context, q *models.Query, res *htt
 	}
 
 	r = processExemplars(q, r)
-	return r, nil
+	return &r, nil
```

Comment:
Change it, by the way, I'm not sure about using an object constructor instead of nil in places like this - https://github.com/grafana/grafana/pull/57209/commits/645c9429ae64d203b309415d23850bdf6dd79285#diff-e3f87c239f58b5cd6879ba3ff6d6ba25b272257650c56b6874d6f35695c1e81aR145, but maybe it's okay

Reply:
same here, i think that's probably fine since the callers should be checking the error. i do have one related open question: https://github.com/grafana/grafana/pull/57209#discussion_r999603684